### PR TITLE
Script Design Audit - created by & created at

### PR DESCRIPTION
**Pull request by Suyash-d**
**Description**
Store the username and timestamp of the last modification of a script. The DES_SCRIPT table should be extended with two new colums:
* CREATED_BY:
  * description: the username of the user that created the script. This information is retrieved from the currently authenticated user when creation is performed via the REST API. Should the script be loaded from the command line then the default 'admin' account is used
  * type: string (255)
  * nullable: false. 
* CREATED_AT:
  * description: the timestamp of creation
  * type: string (255)
  * nullable: false

**Tasks**
1. a
2. alter table script for existing data models

### CODE REVIEW

File: core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java [hunk 1]
```diff
@@ -142,7 +142,9 @@ public Script deserialize(JsonParser jsonParser, DeserializationContext deserial
                     scriptVersion,
                     scriptParameters,
                     scriptActions,
-                    scriptLabels);
+                    scriptLabels,
+                    "createdBy",
+                    "createdAt");
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java:147]: what are these default values?

File: core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java [hunk 2]
```diff
@@ -75,11 +76,15 @@ public Script deserialize(JsonParser jsonParser, DeserializationContext deserial
                 versionNumber = versionNode.get(ScriptVersionJsonComponent.Field.NUMBER_KEY.value()).asLong();
                 scriptVersion = new ScriptVersion(
                         new ScriptVersionKey(new ScriptKey(scriptId, versionNumber)),
-                        versionNode.get(ScriptVersionJsonComponent.Field.DESCRIPTION_KEY.value()).asText()
+                        versionNode.get(ScriptVersionJsonComponent.Field.DESCRIPTION_KEY.value()).asText(),
+                        "username",
+                        LocalDateTime.now().toString()
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java:81]: Where is this used? Isn't this used to load a script from CLI. If so, shouldn't the username be admin?

File: core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java [hunk 3]
```diff
@@ -75,11 +76,15 @@ public Script deserialize(JsonParser jsonParser, DeserializationContext deserial
...
                 );
             } else {
                 versionNumber = 0L;
-                scriptVersion = new ScriptVersion(new ScriptVersionKey(new ScriptKey(scriptId, versionNumber)), "default version");
+                scriptVersion = new ScriptVersion(new ScriptVersionKey(
+                        new ScriptKey(scriptId, versionNumber)), "default version",
+                        "username", LocalDateTime.now().toString());
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java:87]: Where is this used? Isn't this used to load a script from CLI. If so, shouldn't the username be admin?

File: core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/repository/DesignMetadataRepository.java [hunk 4]
```diff
@@ -72,6 +73,9 @@ public void save(Script script) {
                 script.setSecurityGroupKey(publicSecurityGroup.getMetadataKey());
                 script.setSecurityGroupName(publicSecurityGroup.getName());
             }
+
+            script.getVersion().setCreatedBy("admin");
+            script.getVersion().setCreatedAt(LocalDateTime.now().toString());
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/repository/DesignMetadataRepository.java:78]: this causes the creation timestamp to be overriden when updating a script through CLI

Comment (BerrevoetsRobbe) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/repository/DesignMetadataRepository.java:78]: The update of a Script should not modify the created by/at information of a script

File: core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java (hunk 2 above)
Comment (Suyash-d) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java]: Done

File: core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java (hunk 3 above)
Comment (Suyash-d) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/script/ScriptJsonComponent.java]: Done

**Review by BerrevoetsRobbe — COMMENTED** [review]
approved

**Pull request merged**